android/src/main/java/com/mythichelm/localnotifications/LocalNotificationsPlugin.java

```diff
@@ -5,135 +5,132 @@
-    NotificationManager notificationManager =
-            (NotificationManager) getActiveContext().getSystemService(Context.NOTIFICATION_SERVICE);
-    notificationManager.cancel(id);
-  }
-
-  public static boolean handleIntent(Intent intent) {
-    if (intent != null) {
-      String callbackName = intent.getStringExtra(CALLBACK_KEY);
-      String payload = intent.getStringExtra(PAYLOAD_KEY);
-      if (callbackName != null && callbackName != "") {
-        MethodChannel channel  = LocalNotificationsService.getSharedChannel();
-        if (channel != null) {
-          Log.d(LOGGING_TAG, "Invoking method " + callbackName + "('" + payload + "')");
-          channel.invokeMethod(callbackName, payload);
-          return true;
+    private Context getActiveContext() {
+        return (registrar.activity() != null) ? registrar.activity() : registrar.context();
+    }
+
+    @Override
+    public boolean onNewIntent(Intent intent) {
+        return handleIntent(intent);
+    }
+
+    @Override
+    public void onMethodCall(MethodCall call, Result result) {
+        List<Object> arguments = call.arguments();
+        switch (call.method) {
+            case CREATE_NOTIFICATION:
+                result.success(createNotification(arguments));
+                break;
+            case REMOVE_NOTIFICATION:
+                int id = (int) arguments.get(0);
+                removeNotification(id);
+                result.success(null);
+                break;
+            default:
+                result.notImplemented();
+                break;
+        }
+    }
+
+    private int createNotification(List<Object> arguments) {
+        NotificationSettings notificationSettings = notificationSettingsFactory.createFromArguments(arguments);
+        new GenerateLocalNotificationsTask(
+                getActiveContext(),
+                notificationSettings,
+                new NotificationFactory()
+        ).execute();
+        return notificationSettings.Id;
+    }
+
+    private void removeNotification(int id) {
+        NotificationManager notificationManager = getNotificationManager();
+        notificationManager.cancel(id);
+    }
+
+    private NotificationManager getNotificationManager() {
+        return (NotificationManager) getActiveContext().getSystemService(Context.NOTIFICATION_SERVICE);
+    }
+
+    public static boolean handleIntent(Intent intent) {
+        if (intent == null) {
+            Log.d(LOGGING_TAG, "intent was null");
+            return false;
         }
-        else {
-          Log.d(LOGGING_TAG, "MethodChannel was null");
+
+        return checkAndInvokeCallback(intent);
+    }
+
+    private static boolean checkAndInvokeCallback(Intent intent) {
+        String callbackName = intent.getStringExtra(CALLBACK_KEY);
+        String payload = intent.getStringExtra(PAYLOAD_KEY);
+
+        if (isNullOrEmpty(callbackName)) {
+            Log.d(LOGGING_TAG, "callback name was null or empty");
         }
```

Comment:
If callbackName is null or empty, don't we want to return before attempting to invoke callback?

Reply:
yes, seems like I missed a return statement there!